opal/corelib/range.rb

```diff
@@ -236,11 +266,11 @@ def bsearch(&block)
   end
 
   def to_s
-    "#{@begin}#{@excl ? '...' : '..'}#{@end}"
+    "#{@begin.nil? ? '' : @begin}#{@excl ? '...' : '..'}#{@end.nil? ? '' : @end}"
   end
 
   def inspect
-    "#{@begin.inspect}#{@excl ? '...' : '..'}#{@end.inspect}"
+    "#{@begin.nil? ? '' : @begin.inspect}#{@excl ? '...' : '..'}#{@end.nil? ? '' : @end.inspect}"
```

Comment:
These two probably work also with a simple `&&`, e.g.:

```rb
@end && @end.inspect
```

I don't think we should worry about having false here, as I'm quite sure it's not admissible as a value for a range edge.

Reply:
Well, weird as it may look but it is...

```ruby
[8] pry(main)> (true..true)
=> true..true
[9] pry(main)> (false..false)
=> false..
[10] pry(main)> (true..false) # Type mixing happens here and probably that's why it fails.
ArgumentError: bad value for range
from (pry):9:in `__pry__'
[11] pry(main)> 
```